File: BE/routes/company.routes.js
const express = require('express');
const router = express.Router();
const multer = require('multer');
const path = require('path');
const companyController = require('../controllers/company.controller');
const { authenticateToken } = require('../middleware/auth');
const { requireRole } = require('../middleware/role');

const companyStorage = multer.diskStorage({
  destination: (req, file, cb) => cb(null, path.join(__dirname, '../uploads/logos')),
  filename: (req, file, cb) => cb(null, Date.now() + '-' + file.originalname)
});
const uploadLogo = multer({
  storage: companyStorage,
  limits: { fileSize: 5 * 1024 * 1024 }
});

router.get('/profile', authenticateToken, requireRole('company'), companyController.getProfile);
router.put('/profile', authenticateToken, requireRole('company'), uploadLogo.fields([{ name: 'logo', maxCount: 1 }]), companyController.updateProfile);
router.get('/dashboard', authenticateToken, requireRole('company'), companyController.getDashboard);

module.exports = router;

File: BE-Python/routes/company_routes.py
from flask import Blueprint
from middleware.auth import authenticate_token
from middleware.role import require_role
from controllers import company_controller

company_bp = Blueprint('company', __name__)

@company_bp.route('/profile', methods=['GET'])
@authenticate_token
@require_role('company')
def get_profile():
    return company_controller.get_profile()

@company_bp.route('/profile', methods=['PUT'])
@authenticate_token
@require_role('company')
def update_profile():
    return company_controller.update_profile()

@company_bp.route('/dashboard', methods=['GET'])
@authenticate_token
@require_role('company')
def get_dashboard():
    return company_controller.get_dashboard()
